### ISSUE TITLE
Don't flag array destructuring as mutable

### ISSUE THREAD

**Issue by danielnixon** [RebeccaStevens/is-immutable-type#48]
# Bug Report

```typescript
    for (const p of new URL("").searchParams) {
      // Variable should have an immutability of at least "Immutable" (actual: "Mutable"). eslint(functional/prefer-immutable-types)
      const [k, v] = p;
    }
```

As far as I can tell, there is no observably mutable variable here. `k` and `v` are both string, and the array expression that is the destructuring can't be fiddled with.

## Expected behavior

No lint error.

## Actual behavior

Lint error: Variable should have an immutability of at least "Immutable" (actual: "Mutable"). eslint(functional/prefer-immutable-

## Steps to reproduce

See example above.

## Proposed changes

Don't flag array destructuring expressions as mutable? May also apply to object destructuring.

**Comment by RebeccaStevens** [comment]
I created some test for this, but it seems this library is calculating the immutability correctly.
The issue must be in `eslint-plugin-functional` itself.

**Comment by RebeccaStevens** [comment]
Should be fixed in [eslint-plugin-functional@5.0.2](https://github.com/eslint-functional/eslint-plugin-functional/releases/tag/v5.0.2)

**Comment by danielnixon** [comment]
You are fast!
